typespec/radius/v1/radius.tsp

```diff
@@ -0,0 +1,37 @@
+/*
+Copyright 2023 The Radius Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+    
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import "@typespec/openapi";
+
+using OpenAPI;
+
+/* Using Azure.ResoureManager namespace here is a workaround for typespec bug: https://github.com/Azure/typespec-azure/issues/110" */
+namespace Azure.ResourceManager;
+ 
+@doc("Concrete tracked resource types can be created by aliasing this type using a specific property type.")
+@Azure.ResourceManager.Private.armResourceInternal(TProperties)
+@TypeSpec.Http.includeInapplicableMetadataInPayload(false)
+model TrackedResourceRequired<
```

Comment:
Can adding **required** to the properties be enough?

Reply:
Yes....this should give an error if someone specifies no properties

Reply:
I meant like do we need the **Required** in the model name? Why not just **TrackedResource** (with required properties) instead of **TrackedResourceRequired**?

Reply:
TrackedResource defined by arm specs has properties as an optional property which we do not want.